### PR TITLE
[BE] enable ruff rule `Q` from flake8-quotes

### PR DESCRIPTION
**Pull request by XuehaiPan**
Enable [ruff rule `Q`](https://docs.astral.sh/ruff/rules/#flake8-quotes-q) from flake8-quotes. Fixes:

- [avoidable-escaped-quote (Q003)](https://docs.astral.sh/ruff/rules/avoidable-escaped-quote/#avoidable-escaped-quote-q003)
- [unnecessary-escaped-quote (Q004)](https://docs.astral.sh/ruff/rules/unnecessary-escaped-quote/#unnecessary-escaped-quote-q004)

cc @mrshenli @pritamdamania87 @zhaojuanmao @satgera @gqchen @aazzolini @osalpekar @jiayisuse @H-Huang @kwen2501 @awgu @penguinwu @fegin @XilunWu @wanchaol @fduwjj @wz337 @tianyu-l @wconstab @yf225 @chauhang @d4l3k

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -82,6 +82,7 @@ ignore = [
     "SIM116", # Disable Use a dictionary instead of consecutive `if` statements
     "SIM117",
     "SIM118",
+    "Q000",  # please remove after ufmt is enabled globally
```
Comment (XuehaiPan) [inline pyproject.toml:85]: [bad-quotes-inline-string (Q000)](https://docs.astral.sh/ruff/rules/bad-quotes-inline-string/#bad-quotes-inline-string-q000) is not enabled duo to too many changes.

**Comment by Skylion007** [comment]
Unsure about Q001 and Q002 as those fell like things ufmt should take care of. However, Q003 and Q004 definitely seem good to have (assuming the linter is okay with them).

File: pyproject.toml (hunk 1 above)
Comment (Skylion007) [reply pyproject.toml:85]: I feel like Q003/Q004 could be useful but why Q001/Q002?

Comment (XuehaiPan) [reply pyproject.toml:85]: Do you concerning about `ruff` Q001/Q002 can conflict with `black`; or enabling these two rules is redundant?

Comment (Skylion007) [reply pyproject.toml:85]: Both

Comment (XuehaiPan) [reply pyproject.toml]: Removed Q000, Q001, and Q003. Leaving them to `black`.

**Review by ezyang — APPROVED** [review]
This seems fine assuming it doesn't conflict with ufmt. I know black has a preference for a particular quote style when it is not forced but idk the behavior here

**Pull request closed without merging**